tripal_chado/includes/TripalImporter/OBOImporter.inc

```diff
@@ -1063,25 +1070,61 @@ class OBOImporter extends TripalImporter {
             '!url' => $full_url,
           ]), TRIPAL_WARNING);
       }
-      //What should happen with this stuff?
-      $base_iri = $ontology_results['config']['baseUris'][0];
-      $ontologyID = $ontology_results['ontologyId'];
-      $this->baseIRIs[$short_name] = [$ontologyID, $base_iri];
+      # this is good, but i think i need to do the decode
+      foreach ($ontology_results['response']['docs'] as $each ){
+        $obo_id = $each['obo_id'];
+        $defining_ontology= $each['is_defining_ontology'];
+	# first defining_ontology shiould be true, but if it isnt, take the one with the obo_id=$id
+	if ($defining_ontology == 'false' and $obo_id != $id){
+	  continue;
+	}
+        $iri = urlencode(urlencode($each['iri']));
+        $ontologyID=$each['ontology_name'];
+        $type=$each['type']; # property|class
+      }
+
     }
 
     // Next get the term.
-    $iri = urlencode(urlencode($base_iri . $accession));
-    $full_url = 'http://www.ebi.ac.uk/ols/api/ontologies/' . $ontologyID . '/terms/' . $iri;
-    $response = drupal_http_request($full_url, []);
-    if (!$response) {
-      throw new Exception(t('Did not get a response from EBI OLS trying to lookup term: !id',
-        ['!id' => $id]));
-    }
-    $results = drupal_json_decode($response->data);
+    $query = $type;
+    if ($type == 'class'){
+      $query = 'term';
+    }
+    $results =  $this->oboEbiLookup($id,$query);
     if (!$results) {
-      $this->logMessage('Error: no data with !url.  The response was: !response', [
-        '!url' => $full_url,
+      $query = 'query-non-local';
+      $ontology_results = $this->oboEbiLookup($id,$query);
+      if($ontology_results ){
+	  foreach ($ontology_results['response']['docs'] as $each ){
+            $obo_id = $each['obo_id'];
+            $defining_ontology= $each['is_defining_ontology'];
+           if (!$defining_ontology and $obo_id != $id){  
+		   continue;
+            }  
+      	    $iri = urlencode(urlencode($each['iri']));
+            $ontology=$each['ontology_name'];
+            $type=$each['type']; # property|class
+	    if ($type == 'property'){
+              $type = 'properties';
+            }elseif($type == 'class'){
+              $type = 'terms';
+            }
+ 	    $full_url = 'http://www.ebi.ac.uk/ols/api/ontologies/' . $ontology . '/' . $type . '/' . $iri;
+
+            $options = [];
+            $response = drupal_http_request($full_url, $options);
+	    $results = drupal_json_decode($response->data);
```

Comment:
I think these lines should go in the `oboEbiLookup` function  and  in that function, add a new `ifelse` statement for "type".  Then these lines get replaced with the function call.

Reply:
Okay. I created a new ifelse statement and have oboEbiLookup accept a complete iri if one is found in the query. I call the function like this: `$results = $this->oboEbiLookup($id, $type, $found_iri);`